### DISCUSSION TITLE
What could be causing the final status to failed

### DISCUSSION THREAD

**Discussion by isaacporras** [OE4T/meta-tegra#1318] [General]
Hi,
I'm currently flashing my custom carrier board (with Orin NX 8GB SOM) with a machine based on `p3509-a02-p3767-0000`. When I flash using the initrd-flash at the last step it keeps showing **Final status: FAILED** without any apparent reason (all the previous steps finished successfully). I saw in the script code that that status is a cat from `final_status=$(cat $mnt/flashpkg/status)`. I wanted to know what could be causing this error 

`== Step 5: Waiting for final status from device at 2023-08-07T11:32:26-06:00 ==
Waiting for USB storage device flashpkg from 8217f607...[/dev/sdb]
Unmounted /dev/sdb.
Final status: FAILED
Successfully finished at 2023-08-07T11:32:29-06:00
Host-side log:              log.initrd-flash.2023-08-07-10.53.01`

**Comment by madisongh** [comment]
There should be more detail in the logs, particularly the device-side one that should have been copied over during step 5.

**Comment by isaacporras** [comment]
In the device-logs-2023-08-10-11.51.46 folder I can see two files:
-  bootloader.log: is completely empty
- [export-nvme0n1.log](https://github.com/OE4T/meta-tegra/files/12345353/export-nvme0n1.log)
Maybe it is not being able to write the file correctly and that is why it is reporting the error?

**Reply by madisongh** [reply]
The empty bootloader.log doesn't sound good; updating the boot firmware is probably what failed.

One thing to try is to use `./doflash.sh --spi-only` to directly flash the boot firmware into the flash.  If that succeeds, you may need to repeat the initrd flashing, but use `--skip-bootloader` on that step.

**Reply by dwalkes** [reply]
Follow-up thread in https://github.com/OE4T/meta-tegra/discussions/1423
